src/loaders/compute_tools/mash/mash.py

```diff
@@ -0,0 +1,69 @@
+"""
+Run Mash on a set of assemblies.
+"""
+import json
+import threading
+import time
+from pathlib import Path
+from typing import Dict
+
+from src.loaders.compute_tools.tool_common import ToolRunner, run_command
+
+
+def _run_mash_single(
+        genome_id: str,
+        source_file: Path,
+        output_dir: Path,
+        debug: bool,
+        kmer_size: int = 19,
+        sketch_size: int = 10000) -> None:
+    # RUN mash sketch for a single genome
+    command = ['mash', 'sketch',
+               '-o', source_file,  # Output prefix.
+                                   # Save result file to source file directory. The suffix '.msh' will be appended.
+               '-k', f'{kmer_size}',
+               '-s', f'{sketch_size}',
+               source_file]
+
+    run_command(command, output_dir / f'{genome_id}' if debug else None)
+
+    # Save run info to a metadata file in the output directory for parsing later
+    metadata_file = output_dir / f'{genome_id}' / f'mash_run_metadata.json'
+    metadata = {'source_file': str(source_file), 'kmer_size': kmer_size, 'sketch_size': sketch_size}
```

Comment:
yea. I just haven't thought through the best way to do this yet. And we need some additional testing with NERSC.

Reply:
maybe the task runner should be responsible for writing this file? Then you just need the image hash and any arguments to the container that could affect the output (and I'm not sure there are any)

Reply:
we do need source_file info to trace the result file in this case. I think it makes the most sense to generate the metadata here. But task runner can generate some common meta that applies to every execution.

Reply:
Hmm, maybe the task runner adds the docker image hash to the file above then...?

Reply:
maybe. We do have a ticket regarding this. There are some factors that need to be taken into consideration.

Reply:
Ok, we can figure this out later. Maybe the task runner just adds a separate file with all the docker hashes that were run against the source data or something... it might be easier to start moving this stuff to a DB rather than having a zillion files

Reply:
yea I agree.